### DISCUSSION TITLE
Version 0.5

### DISCUSSION THREAD

**Discussion by woosaaahh** [woosaaahh/sj.nvim#7] [Announcements]
Hi everyone !

The version 0.5 was just realeased. 

Please create issues if needed or come here to discuss about this new version ! :smile:

**Comment by jemag** [comment]
hey just wanted to clarify what I mean and reply to : 

> From what I see in your screenshot, you have search_scope = "visible_lines_below" somewhere.
> Try search_scope = "visible_lines" and you'll get what you said. Unless I missed/misunderstood something.

My search scope is set to "buffer". I guess what I meant is that despite being able to search everywhere, it might be more useful to only have labels within the "visible_lines" scope. Since labels that are outside of view will probably not be used anyway.

So essentially ` search_scope = "buffer"` combined with something like `label_scope = visible_lines`. Basically, buffer"  search, but with labeling similar to the current `search_scope = "visible_lines"`. 

 `search_scope = "buffer"` would end up looking like this instead : 
![image](https://user-images.githubusercontent.com/7985687/193355224-67048ea4-2509-4f9c-a821-c3d092448ec1.png)

**Reply by woosaaahh** [reply]
TLDR : So, now I understand but I'm not convinced yet.

With my current config, I have 'run()' keymaps set, one with 'visible_lines' scope and one with 'buffer' scope but I also have 'redo()' keymaps, one with no config and one with `auto_jump=1, relative_labels=false, max_pattern_length=1, search_scope=visible_lines`. The latter is incredibly useful for me.

If I need to see matches out of viewport, I use Ctrl-D/U/F/B and use a 'redo()' keymap or I cycle on prev/next matches after search.

I'll try to think of it more but I'm really not sure if I'll implement this. We'll see !

Even if I'm not convinced by the benefit of what you're asking for this time, that doesn't mean I'm opposed to any suggestions/comment/report/issue. So please, feel free to keep coming and discuss here if needed. I'd really appreciate this !

Until sunday night, I'll try to catch/fix all the bugs I can find in order to merge 0.5 to main.

**Comment by jemag** [comment]
Yeah to be honest it is more of a minor thing. It mostly allows me to almost always use ` search_scope = "buffer", forward_search = true` for almost all my needs (searching through whole buffer and navigating anywhere within current viewport). I could always either do a backward search or do a redo with `search_scope=visible_lines` to accomplish the same thing, it just removes the small mental overhead of having to think of other scenarios or possibly having to re-trigger.

Not sure if there would be other uses cases where having a distinct `search_scope` and  `label_scope` would be useful though.

**Comment by jemag** [comment]
actually nevermind, it would probably conflict with the scenario of having no search item in the current viewport. The more I use it, the more I think I actually don't need this modification.

**Comment by woosaaahh** [comment]
Btw, I found a way to solve a previous issue with 'label_as_prefix` (no overlapping with matches). 
I think I'll re-add this option.
